avocado/core/nrunner/runnable.py

```diff
@@ -64,6 +68,8 @@ class Runnable:
     """
 
     def __init__(self, kind, uri, *args, config=None, **kwargs):
+        if config is None:
+            config = self.essential_config_filter(kind, {})
```

Comment:
Here is something it is still confusing.

We are changing the logic here with the config. Before: full config, Now: only runnable config items.

Users of this constructor will pass a full config or a runnable config? IIUC, we should assume they will start using the "runnable config" (filtered already). I would leave this to the caller of this constructor. We should only warning in case of validation error.

Reply:
So here, maybe you should call the setter, that will call the validator.

Reply:
Yes, I call the setter with validator when there is some config from the constructor caller. So this force them to use "runnable config". But when the config is None, the constructor will set default "runnable config". And that is basically the main enhancement which I would like to introduce here.